### PR TITLE
Add a Karma section to the front-end developer documentation

### PR DESCRIPTION
**Pull request by gorkat**

### CODE REVIEW

File: languages/en/developer-guide/front-end/karma.rst [hunk 1]
```diff
@@ -0,0 +1,109 @@
+Karma unit tests
+================
+
+`Karma <https://karma-runner.github.io/2.0/index.html>`_ is a test runner that we use with the testing framework `Jasmine <https://jasmine.github.io/2.0/introduction.html>`_ to write down our JavaScript unit tests.
```
Comment (Hyzual) [inline languages/en/developer-guide/front-end/karma.rst]: I would prefer pointing to "https://jasmine.github.io/" for jasmine. It seems to not be tied to a version (I have a patch to move to v3)

File: languages/en/developer-guide/front-end/karma.rst [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+You must provide some unit tests for any front-end development.
+
+In this section you will find guidelines on how to setup your own karma tests.
+
+How to bootstrap your unit tests
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In a first place, it is necessary to have a karma configuration file named ``karma.conf.js``.
```
Comment (Hyzual) [inline languages/en/developer-guide/front-end/karma.rst]: In a first place > First

File: languages/en/developer-guide/front-end/karma.rst [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+This config file is pretty easy to set up.
+
+.. code-block:: JavaScript
+
+    // tuleap/plugins/<your_plugin>/scripts/karma.conf.js
```
Comment (Hyzual) [inline languages/en/developer-guide/front-end/karma.rst]: Thanks :)

File: languages/en/developer-guide/front-end/karma.rst [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+    const path = require("path");
+    const webpack_config = require("./webpack.config.js");
+    const karma_configurator = require("../../../../tools/utils/scripts/karma-configurator.js");
```
Comment (Hyzual) [inline languages/en/developer-guide/front-end/karma.rst]: According to the path above, I think there is one ".." too much. Not certain though

File: languages/en/developer-guide/front-end/karma.rst [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        const base_config = karma_configurator.setupBaseKarmaConfig(
+            config,
+            webpack_config,
+            coverage_dir
+        );
+
+        Object.assign(base_config, {
+            files: ["repositories/src/index.spec.js"],      // Here put the relative path to your index.spec.js file.
+            preprocessors: {
+                "repositories/src/app.spec.js": ["webpack"] // Same.
```
Comment (Hyzual) [inline languages/en/developer-guide/front-end/karma.rst]: rename to index.spec.js to match above

File: languages/en/developer-guide/front-end/karma.rst [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+    */
+    import "tlp-mocks";
+
+    // Import all the local tests.
+    import "./store/mutations.spec.js";
+    import "./store/actions.spec.js";
+    import "./store/getters.spec.js";
+    import "./api/reqt-querier.spec.js";
+
+.. note:: All your tests **must** be located in the same folder than the tested files and to be prefixed with ``.spec.js``.
```
Comment (Hyzual) [inline languages/en/developer-guide/front-end/karma.rst]: and -to be -prefixed +suffixed

File: languages/en/developer-guide/front-end/karma.rst (hunk 4 above)
Comment (gorkat) [reply languages/en/developer-guide/front-end/karma.rst]: you are right, it is actually ../../../

File: languages/en/developer-guide/front-end/karma.rst [hunk 7]
```diff
@@ -1,7 +1,7 @@
 Karma unit tests
 ================
 
-`Karma <https://karma-runner.github.io/2.0/index.html>`_ is a test runner that we use with the testing framework `Jasmine <https://jasmine.github.io/2.0/introduction.html>`_ to write down our JavaScript unit tests.
+`Karma <https://karma-runner.github.io/2.0/index.html/>`_ is a test runner that we use with the testing framework `Jasmine <https://jasmine.github.io>`_ to write down our JavaScript unit tests.
```
Comment (Hyzual) [inline languages/en/developer-guide/front-end/karma.rst]: The added slash at the end causes a 404

File: languages/en/developer-guide/front-end/karma.rst [hunk 8]
```diff
@@ -78,10 +78,15 @@ The goal of this file is to import every ``*.spec.js`` file.
...
-.. note:: All your tests **must** be located in the same folder than the tested files and to be prefixed with ``.spec.js``.
+.. note:: All your tests **must** be located in the same folder than the tested files and to be suffixed with ``.spec.js``.
 
 To run your tests locally, use the :ref:`npm scripts <npm_scripts>`.
 
+.. note:: The Karma server might be unable to launch the unit tests if no browser is available. Please ensure that:
+
+    - Chromium is installed on your machine if it runs under Linux.
+    - Chrome is installed otherwise.
+
```
Comment (Hyzual) [inline languages/en/developer-guide/front-end/karma.rst]: :+1:

File: languages/en/developer-guide/front-end/karma.rst (hunk 7 above)
Comment (gorkat) [reply languages/en/developer-guide/front-end/karma.rst]: argh fu :(

File: languages/en/developer-guide/front-end/karma.rst [hunk 9]
```diff
@@ -76,7 +76,7 @@ The goal of this file is to import every ``*.spec.js`` file.
     import "./store/mutations.spec.js";
     import "./store/actions.spec.js";
     import "./store/getters.spec.js";
-    import "./api/reqt-querier.spec.js";
```
Comment (gorkat) [inline languages/en/developer-guide/front-end/karma.rst]: rekt

**Pull request merged**